build_runner/lib/src/generate/create_merged_dir.dart

```diff
@@ -127,13 +115,21 @@ Future<bool> _createMergedOutputDir(
     var packagesFileContent = packageGraph.allPackages.keys
         .map((p) => '$p:packages/$p/')
         .join('\r\n');
-    for (var dir in rootDirs) {
-      var packagesAsset = new AssetId(root, p.url.join(dir, '.packages'));
+
+    if (root != null) {
+      var packagesAsset = new AssetId(root, '.packages');
       _writeAsString(outputDir, packagesAsset, packagesFileContent);
       outputAssets.add(packagesAsset);
-      var link = new Link(p.join(outputDir.path, dir, 'packages'));
-      if (!link.existsSync()) {
-        link.createSync(p.join('..', 'packages'), recursive: true);
+    } else {
+      for (var dir
+          in _findRootDirs(outputPath, assetGraph, packageGraph, buildPhases)) {
+        var packagesAsset = new AssetId(dir, p.url.join(dir, '.packages'));
```

Comment:
Same here, `packageGraph.root.name` instead of `dir`

Reply:
Done.